### PR TITLE
[WIP] Add RestartPolicy to QueueJob

### PR DESCRIPTION
**Pull request by etiennecoutaud**
referering #271

### CODE REVIEW

File: pkg/controller/queuejob/queuejob_controller.go [hunk 1]
```diff
@@ -363,8 +363,23 @@ func (cc *Controller) manageQueueJob(qj *arbv1.QueueJob, pods map[string][]*v1.P
...
+			switch ts.RestartPolicy {
+			case arbv1.RestartPolicyAlways:
+				diff = replicas - pending - running
+			case arbv1.RestartPolicyNever:
+				diff = 0
+			case arbv1.RestartPolicyOnFailure:
+				diff = failed
+			// If no match defaut policy is restart OnFailure
+			default:
+				diff = failed
```
Comment (k82cn) [inline pkg/controller/queuejob/queuejob_controller.go:377]: If no task failed, the diff will be 0; for example, when QueueJob just created.

**Comment by k82cn** [comment]
@etiennecoutaud , I think we may need a design doc for that; for example, what the action we will take if task failed? e.g MPI job.

File: pkg/controller/queuejob/queuejob_controller.go (hunk 1 above)
Comment (etiennecoutaud) [reply pkg/controller/queuejob/queuejob_controller.go:377]: This condition `(running + pending + succeeded) != 0` is for the first time, but I agree for the design doc :)

**Comment by k82cn** [comment]
@etiennecoutaud , I'm going to close this PR for QueueJob, that'll be great if you can contribute PodGroup controller, similar to this PR.

**Comment by etiennecoutaud** [comment]
Sure, I will check `first good issue` label, do you have priority issues to work on ?

**Comment by k82cn** [comment]
for first good issues, you can try the issues that I opened recently :)

**Pull request closed without merging**